openedx/features/content_tagging/rest_api/v1/views.py

```diff
@@ -0,0 +1,181 @@
+
+    **Update Parameters**
+        * id (required): - The id of the taxonomy to update
+
+    **Update Request Body**
+        * name (optional): User-facing label used when applying tags from this taxonomy to Open edX objects.
+        * description (optional): Provides extra information for the user when applying tags from this taxonomy to an object.
+        * enabled (optional): Only enabled taxonomies will be shown to authors.
+        * required (optional): Indicates that one or more tags from this taxonomy must be added to an object.
+        * allow_multiple (optional): Indicates that multiple tags from this taxonomy may be added to an object.
+        * allow_free_text (optional): Indicates that tags in this taxonomy need not be predefined; authors may enter their own tag values.
+
+    **Update Example Requests**
+        PUT api/tagging/v1/taxonomy/:id                                            - Update a taxonomy
+        {
+            "name": "Taxonomy New Name",
+            "description": "This is a new description",
+            "enabled": False,
+            "required": False,
+            "allow_multiple": False,
+            "allow_free_text": True,
+        }
+        PATCH api/tagging/v1/taxonomy/:id                                          - Partially update a taxonomy
+        {
+            "name": "Taxonomy New Name",
+        }
+
+    **Update Query Returns**
+        * 200 - Success
+        * 403 - Permission denied
+
+    **Delete Parameters**
+        * id (required): - The id of the taxonomy to delete
+
+    **Delete Example Requests**
+        DELETE api/tagging/v1/taxonomy/:id                                         - Delete a taxonomy
+
+    **Delete Query Returns**
+        * 200 - Success
+        * 404 - Taxonomy not found
+        * 403 - Permission denied
+
+    """
+    class TaxonomyPagination(DefaultPagination):
+        page_size = 100
+        max_page_size = 1000
+
+    pagination_class = TaxonomyPagination
+
+    class UserOrgFilterBackend(BaseFilterBackend):
+        """
+        Filter taxonomies based on the user's organization.
+        """
+        def filter_queryset(self, request, queryset, view):
+            if is_global_taxonomy_admin(request.user):
+                return queryset
+            else:
+                user_orgs = get_user_taxonomy_orgs(request.user)
+                user_orgs_short_name = [org.short_name for org in user_orgs]
+                return queryset.filter(
+                    Q(
+                        Exists(
+                            TaxonomyOrg.objects.filter(
+                                taxonomy=OuterRef("pk"),
+                                rel_type=TaxonomyOrg.RelType.OWNER,
+                                org__short_name__in=user_orgs_short_name,
+                            )
+                        )
+                    )
+                    | Q(enabled=True)
+                )
+
+    filter_backends = [UserOrgFilterBackend]
```

Comment:
I think this is the way to go. The whole rule/queryset problem is a common issue and is described here: https://open-edx-proposals.readthedocs.io/en/latest/best-practices/oep-0009-bp-permissions.html

Reply:
I'm not sure about the code standards on this. If we think we should move this class to another place, let me know!